Openbao associated data (#4365)
opentofu/opentofu: website/docs/language/state/encryption.mdx

| address                  | OpenBao server address to access the API. OpenTofu can read it from the `BAO_ADDR` environment variable as well. Your system must trust the TLS certificate of the server.  | N/A  | https://127.0.0.1:8200             |
| transit_engine_path      | Path at which the Transit Secret Engine is enabled in OpenBao. Customize this if you changed the transit engine path.                                                       | N/A  | /transit                           |
| key_length               | Number of bytes to generate as a key. Available options are `16`, `32` or `64` bytes.                                                                                       | 16   | 32                                 |

> | associated_data          | Base64-encoded sent with data key generation and decryption.                                                                                                                | N/A  | -                                  |

Reviewer: This documentation is unclear: is the user doing the encoding or is OpenTofu?

> | associated_data          | Must be a base64-encoded string in config (use `base64encode()` for plain text) sent to OpenBao Transit with data key generation and decryption.                            | N/A  | -                                  |

Reviewer 2: ```suggestion
| associated_data          | Base64-encoded string sent to OpenBao Transit with data key generation and decryption providing authenticity protection.                          | N/A  | -                                  |
```
Considering the tone used for the other attributes, I would say that this would be more in line.
